client-rpc/src/rpc/wallet_rpc.rs

```diff
@@ -34,6 +34,15 @@ pub trait WalletRpc: Send + Sync {
     #[rpc(name = "wallet_list")]
     fn list(&self) -> Result<Vec<String>>;
 
+    #[rpc(name = "wallet_listPublicKeys")]
+    fn list_public_keys(&self, request: WalletRequest) -> Result<Vec<PublicKey>>;
```

Comment:
I'd recommend to convert `public_keys` to `String` before returning (similar to what we do before returning addresses).